cmake/CMakeLists.txt

```diff
@@ -332,8 +352,13 @@ if (onnxruntime_USE_TVM)
   add_definitions(-DUSE_TVM)
 
   set(onnxruntime_tvm_libs onnxruntime_codegen_tvm)
-  list(APPEND onnxruntime_EXTERNAL_LIBRARIES tvm nnvm_compiler)
-
+  # needs to link with stdc++fs in Linux
+  if(UNIX)
+    if (NOT APPLE)
+      set(FS_STDLIB stdc++fs)
```

Comment:
Who need this? I just spent a lot of time on getting rid of it.

Reply:
I think some codegen JIT caching is using it, but we might be able to find some alternative.

Reply:
Because MacOSX doesn't have this lib, I'm wondering if nuphar can work on Mac OS X or not. If it can't, we should explicitly mark it at somewhere.

Reply:
I thought it marks not apple.
If the system doesn't have that, nuphar won't cache jit result, meaning it always performs JIT, and getting extra overhead in initialization time.
Nuphar itself seems working in our Mac CI.

Reply:
You are saying, the library itself has some '#ifdef' to avoid using std filesystem on Mac? If you have tested, it's good.

Reply:
Seems like a bug, mac should have FS_STDLIB too.

---
In reply to: [277525483](https://github.com/Microsoft/onnxruntime/pull/881#discussion_r277525483) [](ancestors = 277525483)

Reply:
Nuphar need this to create directory and save file. See onnxruntime/core/providers/nuphar/common/nuphar_tvm_utils.cc

---
In reply to: [277505593](https://github.com/Microsoft/onnxruntime/pull/881#discussion_r277505593) [](ancestors = 277505593)

Reply:
MacOS seems slow to support it. boost is another choice?

Reply:
I would avoid boost if possible. It would bring in a very large dependency.